src/lib.rs

```diff
@@ -67,21 +76,78 @@ impl CipherSuite for DefaultCipherSuite {
     type OprfCs = opaque_ke::Ristretto255;
     type KeGroup = opaque_ke::Ristretto255;
     type KeyExchange = opaque_ke::key_exchange::tripledh::TripleDh;
-    type Ksf = Argon2<'static>;
+    type Ksf = CustomKsf;
 }
 
 #[cfg(feature = "p256")]
 impl CipherSuite for DefaultCipherSuite {
     type OprfCs = p256::NistP256;
     type KeGroup = p256::NistP256;
     type KeyExchange = opaque_ke::key_exchange::tripledh::TripleDh;
-    type Ksf = Argon2<'static>;
+    type Ksf = CustomKsf;
 }
 
 const BASE64: b64::GeneralPurpose = b64::URL_SAFE_NO_PAD;
 
 type JsResult<T> = Result<T, Error>;
 
+#[derive(Default)]
+struct CustomKsf {
+    argon: Argon2<'static>,
+}
+
+impl Ksf for CustomKsf {
+    fn hash<L: ArrayLength<u8>>(
+        &self,
+        input: GenericArray<u8, L>,
+    ) -> Result<GenericArray<u8, L>, InternalError> {
+        let mut output = GenericArray::default();
+        self.argon
+            .hash_password_into(&input, &[0; argon2::RECOMMENDED_SALT_LEN], &mut output)
+            .map_err(|_| InternalError::KsfError)?;
+        Ok(output)
+    }
+}
+
+fn build_argon2_ksf(
+    t_cost: u32,
+    m_cost: u32,
+    parallelism: u32,
+) -> Result<Option<CustomKsf>, Error> {
+    let mut param_builder = ParamsBuilder::default();
```

Comment:
Just leaving this here to think about. What if someone just wanted to change one parameter of the default params? They wouldn't be able to do that with how we are always setting the parameters here vs checking to make sure they have been specified first before setting them.

Not saying we should support specifying only one parameter, that could easily lead to confusion for the user, but could also be convenient for them if they did their research first. 

I guess I would be learning towards how it is now just for that lower chance that users get confused when using this. I'll leave it up to you though